editor/src/components/editor/store/collaborative-editing.ts

```diff
@@ -681,22 +719,51 @@ export async function claimControlOverProject(projectID: string | null): Promise
   if (projectID == null || !isFeatureEnabled('Baton Passing For Control')) {
     return null
   }
+  if (document.hidden) {
+    return false
+  }
 
   const request = claimProjectControl(projectID, collaborationEditor)
   const response = await callCollaborationEndpoint(request)
-  if (response.type === 'CLAIM_PROJECT_CONTROL_RESULT') {
+  if (response.type === 'REQUEST_PROJECT_CONTROL_RESULT') {
+    return response.successfullyClaimed
+  } else {
+    throw new Error(`Unexpected response: ${JSON.stringify(response)}`)
+  }
+}
+
+export async function snatchControlOverProject(projectID: string | null): Promise<boolean | null> {
+  if (projectID == null || !isFeatureEnabled('Baton Passing For Control')) {
+    return null
+  }
+
+  const request = snatchProjectControl(projectID, collaborationEditor)
+  const response = await callCollaborationEndpoint(request)
```

Comment:
would it make sense to put this into a `try` and handle the failure?

Reply:
Fixed in 3d682504d.